### DISCUSSION TITLE
Function of MagickDeskewImage cannot exit

### DISCUSSION THREAD

**Discussion by lanyangyang845** [ImageMagick/ImageMagick#2158] [Help]
When I use the function of MagickDeskewImage in Ubuntu, my program stuck and cannot exit.
The code is as follows:
MagickWand* ImageConvertFunction::imageDeskew(MagickWand *wand)
{
    MagickBooleanType
            status;
    status = MagickDeskewImage(wand,50000);
    if (status == MagickFalse)
        ThrowWandException(wand);
    return wand;
}

**Comment by urban-warrior** [comment]
Try deskew from the command-line:

`magick myImage.jpg -deskew 50000 null:`

Does the command complete in a reasonable time?  If not, post the a link to your image here.  We need to reproduce the problem before we can offer a solution.  For the image set we tried, deskewing completed without complaint with sub-second response times.

**Comment by lanyangyang845** [comment]
In the operating system I use, there is no command of "magick";
I try the code as follows:
    ReferenceFilename ="res.jpg";
    char * argv[5] = {"convert", "-deskew", "50000", "test.jpg", ReferenceFilename}; 
    CopyMagickString(image_info->filename,ReferenceFilename,strlen(ReferenceFilename)+1);
    ConvertImageCommand(image_info,5,argv,(char **) NULL,ep);
The above code can be quit when it is executed for the first time, but still stuck when it is executed for the second time. And the image does not refer to a specific one, every one will be like this.
